x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/AbstractProfilingPersistenceManager.java

```diff
@@ -71,6 +73,18 @@ public void setTemplatesEnabled(boolean templatesEnabled) {
         this.templatesEnabled = templatesEnabled;
     }
 
+    private static boolean isMixedVersionCluster(DiscoveryNodes nodes) {
+        Version version = null;
+        for (var n : nodes) {
+            if (version == null) {
+                version = n.getVersion();
+            } else if (version.equals(n.getVersion()) == false) {
+                return false;
```

Comment:
Shouldn't this return `true`? i.e. if the there is a version mismatch in any two nodes we infer a mixed version cluster?

Reply:
Doh, wrong way round...